src/Microsoft.OpenApi/Services/OpenApiWorkspace.cs

```diff
@@ -176,7 +205,7 @@ public bool RegisterComponentForDocument<T>(OpenApiDocument openApiDocument, T c
                 IOpenApiExample => baseUri + ReferenceType.Example.GetDisplayName() + ComponentSegmentSeparator + id,
                 IOpenApiHeader => baseUri + ReferenceType.Header.GetDisplayName() + ComponentSegmentSeparator + id,
                 IOpenApiSecurityScheme => baseUri + ReferenceType.SecurityScheme.GetDisplayName() + ComponentSegmentSeparator + id,
-                _ => throw new ArgumentException($"Invalid component type {componentToRegister.GetType().Name}"),
+                _ => throw new ArgumentException($"Invalid component type {componentToRegister?.GetType().Name}"),
```

Comment:
no need for null prop here, we're already checking against the component being null earlier

Reply:
The compiler still raises a warning on dereference of a possible null reference even with the null check in place